apps/rush/src/RushCommandSelector.ts

```diff
@@ -16,9 +16,10 @@ type CommandName = 'rush' | 'rushx' | undefined;
  */
 export class RushCommandSelector {
   public static failIfNotInvokedAsRush(version: string): void {
```

Comment:
I'm having trouble following this logic now.  Before this PR, the caller's logic looked like:

```ts
    if (semver.lt(version, '3.0.20')) {
      // In old versions, requiring the entry point invoked the command-line parser immediately,
      // so fail if "rushx" was used
      RushCommandSelector.failIfNotInvokedAsRush(version);
      require(path.join(expectedRushPath, 'node_modules', '@microsoft', 'rush', 'lib', 'rush'));
    } else if (semver.lt(version, '4.0.0')) {
```

In other words, if `version` is too old, then `rushx` isn't available, so we "fail if not invoked as `rush`".  

However now there two commands `rushx` and `rush-pnpm`, so the concept seems like it should be more like "fail if invoked using an unimplemented name".  And the version range for `rush-pnpm` is different from the version range for `rushx`.

Seems like `ensureRushVersionInstalledAsync()` might need some changes, and maybe `failIfNotInvokedAsRush()` is not a good abstraction any more. (?)